### ISSUE TITLE
Flaky spec: Admin budget investments Edit Adds existing valuation tags

### ISSUE THREAD

**Issue by aitbw** [consul/consul#2557]
# What

Tests that fail randomly are called "flakies", this one seems to be one:

**Randomized seed:** 52021

**Travis failed build:** [Travis build 20903, job 1](https://travis-ci.org/consul/consul/jobs/359034088)

**Failure:**

```
  1) Admin budget investments Edit Adds existing valuation tags
     Failure/Error: expect(page).to have_content 'Investment project updated succesfully.'
       expected to find text "Investment project updated succesfully." in "Language: English Español Français Nederlands Português CONSUL ADMINISTRATION Admin Notifications My activity My account Sign out Categories Moderated content Polls Participatory budgets Profiles Collaborative Legislation Banners Site customization Manage geozones Signature Sheets Sending of emails Go back dolorem 179 - Group: Group 100 - Investment project 2 Budget Investment 64 title Heading: Heading 96 By: Manuela626 Sent: 2018-03-27 20:55:12 Compatibility: Compatible Selection: Not selected Winner: No Spend money on this Image: Without image Documents: Without documents User tags: Edit Classification Assigned administrator: Undefined Tags: Education Assigned valuators: Undefined Edit classification Dossier Price (€): 10 Cost during the first year (€): Undefined Time scope: Undefined Feasibility: Undecided Edit dossier Valuation comments Leave your comment Comment as admin Milestone Don't have defined milestones Create new milestone"
     # ./spec/features/admin/budget_investments_spec.rb:595:in `block (3 levels) in <top (required)>'
```

# How
- [ ] Explain why the test is flaky, or under which conditions/scenario it fails randomly
- [ ] Explain why your PR fixes it

### Random values issues
If the problem comes from randomly generated values, running multiple times a single spec could help you reproduce the failure by running at your command line:
```bash
for run in {1..10}
do
  bin/rspec ./spec/features/budgets/investments_spec.rb:256
done
```

You can also try running a single spec in Travis:
Add option `:focus` to the spec and push your branch to Github, for example:
```ruby
scenario 'Show', :focus do
```

But remember to remove that `:focus` changes afterwards when submitting your PR changes!

### Test order issues
Running specs in the order they failed may discover that the problem is that a previous test sets an state in the test environment that makes our flaky fail/pass. Tests should be independent from the rest.

After executing rspec you can see the seed used, add it as an option to rspec, for example:
`bin/rspec --seed 55638` (check **Randomized seed** value at beginning of issue)

### Other things to watch for
- Time related issues (current time, two time or date comparisons with miliseconds/time when its not needed)
- [How to Deal With and Eliminate Flaky Tests - Semaphore](https://semaphoreci.com/community/tutorials/how-to-deal-with-and-eliminate-flaky-tests)

### Final thoughts
The true objective of this issue is not "to fix" a flaky spec, but to change a spec that randomly fails into one that we can trust when it passes (or fails). 

That means you've to think "What are we testing here?" and "Can we test the same thing in another way?" or "Are we already testing this somewhere else at least partially?".

Sometimes the fix is to re-write the entire tests with a different approach, or to extend an existing spec. Tests are sometimes written without taking a look at other tests files neither near scenarios.

### To consider

* This flaky is only occurring in CONSUL, so you might check [Decide Madrid](http://github.com/AyuntamientoMadrid/consul) specs as a start

* This issue came up first in #2534

**Comment by javierv** [comment]
I forgot to mention I haven't reproduced this one. Just in case it helps, there's a CSS animation, and the error message says the page content should be exactly what it is (the tag was updated properly) except for the flash message, which is animated by CSS.

So there's an option it's fixed by #2696, or it could already be fixed by something else. It's impossible to be sure without reproducing it locally, but I hope this comment helps anyway!

**Comment by javierm** [comment]
Closing this one since we haven't seen this failure for months and there's a chance it was related to CSS animations.
